### PR TITLE
DBZ-743 Empty password for PostgreSQL and MySQL

### PR DESCRIPTION
**Pull request by jpechane**
https://issues.jboss.org/browse/DBZ-743

### CODE REVIEW

**Comment by gunnarmorling** [comment]
@jpechane The test failure in the MySQL module looks related; this assertion in `MySqlConnectorIT.shouldFailToValidateInvalidConfiguration()` fails:

    assertConfigurationErrors(result, MySqlConnectorConfig.PASSWORD, 1);

File: debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/MySqlConnectorConfig.java [hunk 1]
```diff
@@ -556,7 +556,6 @@ public static DdlParsingMode parse(String value, String defaultValue) {
                                               .withType(Type.PASSWORD)
                                               .withWidth(Width.SHORT)
                                               .withImportance(Importance.HIGH)
-                                              .withValidation(Field::isRequired)
```
Comment (gunnarmorling) [inline debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/MySqlConnectorConfig.java]: So can one have an empty password in MySQL?

File: debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PostgresConnector.java [hunk 2]
```diff
@@ -85,6 +85,10 @@ public Config validate(Map<String, String> connectorConfigs) {
         ConfigValue userValue = results.get(PostgresConnectorConfig.USER.name());
         ConfigValue passwordValue = results.get(PostgresConnectorConfig.PASSWORD.name());
 
+        if (passwordValue.value() == null || ((String)passwordValue.value()).isEmpty()) {
+            logger.warn("The connection password is empty, this is valid setting only for special cases");
```
Comment (gunnarmorling) [inline debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PostgresConnector.java]: I'm a bit torn on that one. We shouldn't lecture people in our logs without good reason :) How about removing the second part of the sentence (after the comma)?

File: debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/MySqlConnectorConfig.java (hunk 1 above)
Comment (jpechane) [reply debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/MySqlConnectorConfig.java]: @gunnarmorling Not sure I undertand the question.

Comment (gunnarmorling) [reply debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/MySqlConnectorConfig.java]: Sorry; what I meant is this: is it legit to remove this validation for MySQL, or is a non-empty password actually required here and only allowed for Postgres? Does that make sense?

Comment (jpechane) [reply debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/MySqlConnectorConfig.java]: @gunnarmorling I envisioned exatcly the same case, you have smart proxy in front of your database when you do not need to proved the credentials. I think we will see more of that when service mesh soltuions like Istio will become more common.

Comment (gunnarmorling) [reply debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/MySqlConnectorConfig.java]: Ok then.

**Comment by gunnarmorling** [comment]
Rebased and applied. Thanks, @jpechane!

**Pull request merged**